ch_pipeline/hfb/flagging.py

```diff
@@ -348,6 +345,61 @@ def process(self, rfimaskbitmap):
         return out
 
 
+class RFIMaskReduceBeamNS(task.SingleTask):
+    """Create RFIMask from HFBDirectionalRFIMaskBitmap.
+
+    This task takes an HFBDirectionalRFIMaskBitmap(freq, beam_ns, time), selects the RFI mask corresponding to a specified significance value used in the detection and subfrequency threshold, then reduce the 'beam_ns' axis to create a RFIMask container (freq, time).
+
+    Attributes
+    ----------
+    sigma : int
+        Specify the value of significance used in RFI detection to create a RFI mask.
+        Must match one of the sigma keys used when the HFBDirectionalRFIMaskBitmap was
+        created. Default is 5.
+    subfreq_threshold : int
+        Subfrequency threshold for decoding the bitmap. A sample is flagged if
+        the number of HFB subfrequency channels detecting RFI >= this threshold.
+        Default is 2.
+    beam_ns_threshold : int
+        This number determines the minimum number of detected RFI events along the beam_ns axis required for a data point
+        to be included in the reduced mask. Default is 1.
+    """
+
+    sigma = config.Property(proptype=int, default=5)
+    subfreq_threshold = config.Property(proptype=int, default=2)
+    beam_ns_threshold = config.Property(proptype=int, default=1)
+
+    def process(self, rfimaskbitmap):
+        """Produce a RFI mask.
+
+        Parameters
+        ----------
+        rfimaskbitmap : containers.HFBDirectionalRFIMaskBitmap
+            beam_ns-specific RFI mask with axes (freq, beam_ns, time).
+
+        Returns
+        -------
+        out : draco.containers.RFIMask(freq, time)
+            Non beam_ns-specific RFI mask with axes (freq, time).
+
+        """
+        # Extract mask/frac and axes data
+        mask = rfimaskbitmap.get_mask(self.sigma, self.subfreq_threshold)[:]
```

Comment:
not being familar with the framework: "[:]" actually does something (forces things being converted to a numpy array or so?)?